packages/google_sign_in/lib/google_sign_in.dart

```diff
@@ -234,7 +265,12 @@ class GoogleSignIn {
   /// a Future which resolves to the same user instance.
   ///
   /// Re-authentication can be triggered only after [signOut] or [disconnect].
-  Future<GoogleSignInAccount> signIn() => _addMethodCall('signIn');
+  Future<GoogleSignInAccount> signIn() {
+    final Future<GoogleSignInAccount> result = _addMethodCall('signIn');
+    bool isCanceled(dynamic error) =>
+        error is PlatformException && error.code == kSignInCanceledError;
+    return result.catchError((dynamic _) => null, test: isCanceled);
```

Comment:
Would it be possible to return `null` from the platform side when the user cancels the sign-in dialog, to avoid reporting that situation as an error wrapped in a `PlatformException` that we then have to catch and investigate here?

I understand that we want to make the information that the user canceled the dialog available to the Dart side, but is there any reason that could not be done by returning `null`? Are there other situations in which we'd return `null` from `signIn`?

Reply:
>  Are there other situations in which we'd return `null` from `signIn`?

Not to my knowledge, however I'm still learning all the intricacies of native plugins and personally do not have a lot of trust in my current understanding of things.

I left this logic on the Dart side because this way it's a bit more explicit. Users can just click through the Dart code and see in which cases `signIn` returns null. It would also be easier to extend in case "canceled" errors would ever need to be exposed explicitly.

Previously in both native implementations we had a single block with `if` statement checking for both `sign_in_required` and `sign_in_canceled`. It was a bit ambiguous in which case which error occur.

That said, I can move it back to the native side if you think it's best handled there. Please let me know your thoughts on this!

Reply:
Thanks for explaining the rationale. I think we should leave it as it is now.